fdbserver/DataDistribution.actor.cpp

```diff
@@ -1849,15 +1901,16 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 				ASSERT(serverTeam.size() == configuration.storageTeamSize);
 
 				std::sort(serverTeam.begin(), serverTeam.end());
-				if (teamExists(serverTeam)) {
+				int overlap = overlappingMembers(serverTeam);
```

Comment:
I agree with the idea of preferring a new team with less overlapped members.

This may not be enough to achieve the goal. When a cluster is small, it may have to create a lot of overlapping teams to meet the target team number. When the cluster is expanded, those overlapping teams are still there, although the new teams will have less overlapping members. 

I think we also want to have a background actor to evaluate the teams and remove teams with a lot of overlapping members.

Reply:
The point is not that overlap is bad, its that you need enough non-overlap to get good distribution. When a cluster is small you will have the overlap, but also as you add more servers those original servers will have proportionally more total teams.